libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/PushNotificationManager.swift

```diff
@@ -84,3 +89,77 @@ extension PushNotificationManager {
+        let response = try await client.send(request: request)
+        let result = try response.asDecodable(type: NotificationTypesResponse.self)
+        return result.notificationTypes
+    }
+    
+    private func storeNotification(types: [NotificationType], with account: UserAccount) {
+        account.notificationTypes = types
+    }
+    
+    private func getCachedNotificationTypes(with account: UserAccount) -> [NotificationType]? {
+        return account.notificationTypes
+    }
+    
+        private func setNotificationCategories(types: [NotificationType]) {
+            let categories = types.map { createNotificationCategory(from: $0) }
+            UNUserNotificationCenter.current().setNotificationCategories(Set(categories))
+        }
+    
+        private func createNotificationCategory(from type: NotificationType) -> UNNotificationCategory {
+            let actions = createActions(from: type.actionGroups)
+            return UNNotificationCategory(identifier: type.apiName, actions: actions, intentIdentifiers: [])
+        }
+    
+        private func createActions(from actionGroups: [ActionGroup]) -> [UNNotificationAction] {
+            return actionGroups.flatMap { actionGroup in
+                actionGroup.actions.map { action in
+                    UNNotificationAction(
+                        identifier: action.identifier,
+                        title: action.label,
+                        options: [.foreground] // Ensures the app opens if needed
+                    )
+                }
+            }
+        }
+}
+
+@objc
+extension PushNotificationManager {
+    /// Retrieves available notification types from the API or local cache if the API request fails.
+    ///
+    /// - Parameters:
+    ///   - restClient: The `RestClient` instance used to fetch notification types from the API.
+    ///   - userDefaults: Stores and retrieves cached notification types. Defaults to `.standard`.
+    /// - Returns: A tuple containing:
+    ///   - `Bool`: `true` if notification types were successfully retrieved and stored; `false` if retrieval failed.
+    ///   - `Error?`: An error object if retrieval fails completely, or `nil` if successful.
+    ///
+    /// If the API fetch fails, the function attempts to retrieve notification types from the cache.
+    /// If neither source provides data, the function returns `false` with an appropriate error.
+    @objc
+    public func getNotificationTypes(restClient: RestClient = RestClient.shared,
+                                     account: UserAccount) async -> (Bool, Error?) {
+        do {
+            let types = try await fetchNotificationTypesFromAPI(with: restClient)
+            storeNotification(types: types, with: account)
+            setNotificationCategories(types: types)
+            return (true, nil)
+        } catch {
+            print("API fetch failed: \(error.localizedDescription). Trying cache...")
```

Comment:
You can use [SFSDKCoreLogger](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKCoreLogger.h) instead of `print`